src/runtime/slice.go

```diff
@@ -54,29 +36,21 @@ func sliceCopy(dst, src unsafe.Pointer, dstLen, srcLen uintptr, elemSize uintptr
 
 // sliceGrow returns a new slice with space for at least newCap elements
 func sliceGrow(oldBuf unsafe.Pointer, oldLen, oldCap, newCap, elemSize uintptr) (unsafe.Pointer, uintptr, uintptr) {
-
-	// TODO(dgryski): sliceGrow() and sliceAppend() should be refactored to share the base growth code.
-
 	if oldCap >= newCap {
 		// No need to grow, return the input slice.
 		return oldBuf, oldLen, oldCap
 	}
 
-	// allow nil slice
-	if oldCap == 0 {
-		oldCap++
-	}
-
-	// grow capacity
-	for oldCap < newCap {
-		oldCap *= 2
-	}
+	// This can be made more memory-efficient by multiplying by some other constant, such as 1.5,
```

Comment:
It's 2x only for small slices (and then toward 1.25x for big ones) in big go - should it also be same for tinygo or expectation is always "small"-ish in tinygo ?

https://github.com/golang/go/blob/e76353d5a923dbc5e22713267104b56a2c856302/src/runtime/slice.go#L304

Reply:
Good idea!  Doing 1.5x for slices over 64 elements now.